prover/src/constraints/evaluator.rs

```diff
@@ -273,13 +265,11 @@ impl<'a, A: Air, E: FieldElement<BaseField = A::BaseField>> ConstraintEvaluator<
         // the results into evaluations buffer
         self.air.evaluate_transition(main_frame, periodic_values, evaluations);
 
-        // merge transition constraint evaluations into a single value and return it;
-        // we can do this here because all transition constraints have the same divisor.
-        self.transition_constraints.main_constraints().iter().fold(E::ZERO, |result, group| {
-            let (power, offset_exp) = (group.degree_adjustment(), group.domain_offset_exp());
-            let xp = domain.get_ce_x_power_at(step, power, offset_exp);
-            result + group.merge_evaluations(evaluations, xp)
-        })
+       // merge constraint evaluations for the main trace segment
+        let result = evaluations.iter().zip(self.transition_constraints.main_constraint_coef().iter()).fold(E::ZERO, |acc, (&const_eval, &coef)| {
+            acc + coef.mul_base(const_eval)
+        });
+        result
```

Comment:
minor nit: I'd improve formatting here - maybe run `rustfmt` manually as it doesn't run for this function by default.

Also:
- I'd keep the previous comment.
- I don't think we need to introduce the `result` variable. We can probably return the result of folding directly.